app/code/Magento/Catalog/Model/ProductRepository.php

```diff
@@ -251,6 +251,11 @@ public function get($sku, $editMode = false, $storeId = null, $forceReload = fal
         if (!isset($this->instances[$sku])) {
             $sku = trim($sku);
         }
+
+        if (!isset($this->instances[$sku])) {
+            throw new NoSuchEntityException(__('Requested product doesn\'t exist'));
```

Comment:
This code completely duplicates previous condition.

What is this issue all about? I see a `trim` but don't see any `strtolower` in logic like `$this->instances[$product->getSku()][$cacheKey] = $product;`.

Reply:
@orlangur, ok, let me explain. We have product with sku "Test". We try to use "get" method of repository to get product with sku "test". Method resourceModel->getIdBySku($sku) will return to us id of "Test" and in $this->instances it put the sku "Test". So after trim we are checking if we have $this->instances["test"] and we don't have it, because we have $this->instances["Test"].

Reply:
How? https://3v4l.org/smbkn

Reply:
@orlangur, what how? Trimming isn't my logic). It was before my changes. I only check if product sku("Test") is the same as requested sku("test"). And if it isn't the same exception will be thrown.

Reply:
Ok, let me reword. Why product is stored with lower-cased SKU in the first place?

We must find a root cause so that exception will be thrown from line 240 instead of adding more exceptions throwing.